### DISCUSSION TITLE
query & mutations make different requests width different ETag

### DISCUSSION THREAD

**Discussion by homerboy19520** [vendurehq/vendure#1676] [Q&A]
Hi!
I try "login()" mutation. Response successful (showed email), in Response Headers get Set-Cookie and server saved it in the browser.
Ok, try checked successful login
Try query "me{identifier}" (can show email), and note that value in ETag in headers different. Response  identifier - undefined

I think the problem is that my requests are not related and they are separate
Also in the UI landing page for requests - after the first one, cookies were saved (like mine), and with the following requests it was automatically added, this cookie is in the request header (not like mine) and ETag does not change

How should I link my queries

**Comment by michaelbromley** [comment]
What client are you using to make requests? The browser will automatically handle cookies for you, as you have seen in the Admin UI. If you are using some other kind of client (postman etc) then you may need to change some setting to enable the session cookie.

**Comment by homerboy19520** [comment]
1. Use Apollo vue
2. In settings UI found "request.credentials": "include", when I set the same value, I got an error in network query 
- Cors Error, fetch, and in request headers - Provisional headers are shown

**Reply by michaelbromley** [reply]
If the server and storefront are not on the same domain, you can run into [SameSite cookie issues](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite). You may be able to work around this by using the `authOptions.cookieOptions` in your vendure config. Failing that, I'd recommend you try implementing the bearer token method of managing sessions.
